File: app/src/main/java/com/coinninja/coinkeeper/view/button/CopyToBufferButton.java
package com.coinninja.coinkeeper.view.button;

import android.content.ClipboardManager;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatButton;

import com.coinninja.coinkeeper.R;
import com.coinninja.coinkeeper.util.android.ClipboardUtil;

public class CopyToBufferButton extends AppCompatButton {

    private ClipboardUtil util;

    public CopyToBufferButton(Context context) {
        super(context);
        init();
    }

    public CopyToBufferButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public CopyToBufferButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        util = new ClipboardUtil((ClipboardManager) getContext().getSystemService(Context.CLIPBOARD_SERVICE));
        super.setOnClickListener(v -> copyClip());
    }

    @Override
    public void setOnClickListener(@Nullable OnClickListener l) {
        return;
    }

    private void copyClip() {
        String clipLabel = getClipLabel();
        util.setClipFromText(clipLabel, getClip());
        Toast.makeText(getContext(), clipLabel, Toast.LENGTH_LONG).show();
    }

    private String getClipLabel() {
        return getContext().getString(R.string.copied, getClip());
    }

    private String getClip() {
        return asString();
    }

    public String asString() {
        return getText().toString();
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/view/button/CopyToBufferButton.kt
package com.coinninja.coinkeeper.view.button

import android.content.ClipboardManager
import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.Toast
import androidx.appcompat.widget.AppCompatButton
import com.coinninja.coinkeeper.R.string
import com.coinninja.coinkeeper.util.android.ClipboardUtil

class CopyToBufferButton : AppCompatButton {
    private var util: ClipboardUtil? = null

    constructor(context: Context?) : super(context) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        util = ClipboardUtil(context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager)
        super.setOnClickListener { v: View? -> copyClip() }
    }

    override fun setOnClickListener(l: OnClickListener?) {
        return
    }

    private fun copyClip() {
        val clipLabel = clipLabel
        util!!.setClipFromText(clipLabel, clip)
        Toast.makeText(context, clipLabel, Toast.LENGTH_LONG).show()
    }

    private val clipLabel: String
        private get() = context.getString(string.copied, clip)

    private val clip: String
        private get() = asString()

    fun asString(): String {
        return text.toString()
    }
}